module/VuFind/src/VuFind/Controller/OverdriveController.php

```diff
@@ -0,0 +1,311 @@
+<?php
+/**
+ * Overdrive Controller
+ *
+ * PHP version 7
+ *
+ * @category VuFind
+ * @package  Controller
+ * @author   Brent Palmer <brent-palmer@ipcl.org>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org Main Site
+ */
+namespace VuFind\Controller;
+
+use Zend\Log\LoggerAwareInterface;
+use Zend\ServiceManager\ServiceLocatorInterface;
+
+/**
+ * Overdrive Controller supports actions for Overdrive Integration
+ *
+ * @category VuFind
+ * @package  Controller
+ * @author   Brent Palmer <brent-palmer@ipcl.org>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org/wiki/development:plugins:controllers Wiki
+ */
+class OverdriveController extends AbstractBase implements LoggerAwareInterface
+{
+    use \VuFind\Log\LoggerAwareTrait {
+        logError as error;
+    }
+
+    /**
+     * Overdrive Connector
+     *
+     * @var \VuFind\DigitalContent\OverdriveConnector $connector Overdrive Connector
+     */
+    protected $connector;
+
+    /**
+     * Constructor
+     *
+     * @param ServiceLocatorInterface $sm Service locator
+     */
+    public function __construct(ServiceLocatorInterface $sm)
+    {
+        $this->setLogger($sm->get('VuFind\Logger'));
+
+        $this->connector = $sm->get('VuFind\DigitalContent\OverdriveConnector');
+        parent::__construct($sm);
+        $this->debug("ODRC constructed");
+    }
+
+    /**
+     * My Content Action
+     * Prepares the view for the Overdrive MyContent template.
+     *
+     * @return array|bool|\Zend\View\Model\ViewModel
+     */
+    public function mycontentAction()
+    {
+        $this->debug("ODC mycontent action");
+
+        //TODO get hold and checkoutlimit using the Patron Info API
```

Comment:
Just a placeholder for future functionality.  I can remove it if you prefer.

Reply:
Feel free to leave them if they'll be useful for future reference... but I'd suggest reviewing all of your TODO's throughout the PR in case there are any that are obsolete, just so they don't confuse us in the future.

Reply:
I've reviewed them.  I changed some of the Todos to something else like "future" or something and removed some.